### PR TITLE
Update to use `windows-latest` as the build agent image

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Update to use `windows-latest` as the build agent image for CI, because the windows-2016 is being deprecated.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

**Review by PaulHigin — APPROVED** [review]

**Review by iSazonov — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
There are 3 other files that need to be updated: windows-test.yml, windows-packaging.yml, PSGalleryToAzArtifiacts.yml

**Review by adityapatwardhan — APPROVED** [review]

**Review by PaulHigin — APPROVED** [review]

File: test/powershell/Modules/Microsoft.WSMan.Management/CredSSP.Tests.ps1 [hunk 1]
```diff
@@ -44,50 +47,25 @@ Describe "CredSSP cmdlet tests" -Tags 'Feature','RequireAdminOnWindows' {
...
-            $c[0] | Should -Match "The machine is configured to allow delegating fresh credentials to the following target\(s\):wsman/\*"
+            Wait-UntilTrue -IntervalInMilliseconds 500 -sb {
+                $c = Get-WSManCredSSP
+                $c[0] -match $expected
+            } | Should -BeTrue -Because "WSManCredSSP should have been enabled to allow delegating fresh credentials to wsman/*, but it was not."
         }
         else
         {
-            $c[1] | Should -Match "This computer is configured to receive credentials from a remote client computer"
+            $c = Get-WSManCredSSP
```
Comment (PaulHigin) [inline test/powershell/Modules/Microsoft.WSMan.Management/CredSSP.Tests.ps1:67]: The delay is only for 'client' role?  It seems like we should be safe and use 'Wait-UntilTrue' for the 'server' role as well.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: test/powershell/Modules/Microsoft.WSMan.Management/CredSSP.Tests.ps1 (hunk 1 above)
Comment (daxian-dbw) [reply test/powershell/Modules/Microsoft.WSMan.Management/CredSSP.Tests.ps1]: Make senses. I will update this branch too.

**Review by PaulHigin — APPROVED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by TravisEz13** [comment]
https://github.com/PowerShell/PowerShell/commit/75205b8096be1d0a345dfac5904ef6110b359c01

**Pull request merged**